interface.py

```diff
@@ -0,0 +1,258 @@
+from models.Budget import Budget
+from models.Category import Category
+from models.ParentCategory import ParentCategory
+from models.Transaction import Transaction
+from models.User import User
+from controllers.budget_controller import add_new_budget
+from session import session
+
+from prettytable import PrettyTable
+
+
+def menu():
+    print("MENU:")
+    print("0. Change user")
+    print("1. Change budget")
+    print("2. New transaction (activity)")
+    print("3. Edit categories")
+    print("4. Edit budget (modify budgeted amounts)")
+    print("5. Switch the month (the billing peroid")
+    print("6. Reports")
+    print("7. Save and close the program")
+    choice = input("## YOUR CHOICE: ")
+
+    if choice == "0":
+        change_user()
+    if choice == "1":
+        change_budget()
+    if choice == "2":
+        new_transaction()
+    if choice == "3":
+        edit_categories()
+    if choice == "4":
+        edit_budget()
+    if choice == "5":
+        switch_month()
+    if choice == "6":
+        reports()
+    if choice == "7":
+        print("End. Program closed.")
+        exit()
+    else:
+        print("\n!!! >>> WRONG CHOICE (OUT OF RANGE). TRY AGAIN...")
+        menu()
```

Comment:
The function shouldn't call itself like that. If someone holds "Enter" key they will hit a recursion limit of 1000 pretty quickly and the program will crash.


```
File "/mnt/d/infoshare/git/jpydzr2-snaver/main.py", line 25, in <module>
    import controller
  File "/mnt/d/infoshare/git/jpydzr2-snaver/controller.py", line 61, in <module>
    menu(global_user_id)
  File "/mnt/d/infoshare/git/jpydzr2-snaver/controller.py", line 53, in menu
    menu(global_user_id)
  File "/mnt/d/infoshare/git/jpydzr2-snaver/controller.py", line 53, in menu
    menu(global_user_id)
  File "/mnt/d/infoshare/git/jpydzr2-snaver/controller.py", line 53, in menu
    menu(global_user_id)
  [Previous line repeated 448 more times]
  File "/mnt/d/infoshare/git/jpydzr2-snaver/controller.py", line 28, in menu
    choice = input("## YOUR CHOICE: ")
```

Reply:
Fixed